fud2/fud-core/src/flang/ir.rs

```diff
@@ -77,10 +91,114 @@ impl Ir {
         self.paths.push(path.clone())
     }
 
+    pub fn path_ref_of_str(&mut self, path: &str) -> PathRef {
+        for (r, p) in &self.paths {
+            if p == path {
+                return r;
+            }
+        }
+        self.paths.push(path.into())
+    }
+
     pub fn path(&self, r: PathRef) -> &Utf8PathBuf {
```

Comment:
Should this maybe return a `&Utf8Path` instead of a buffer reference? Seems a little less restrictive, and I'm not sure we ever need to know that we're handing out references to entire (mutable) buffers…

(Apologies if I'm stating the obvious, but: `&Utf8Path` is to `&str` as `Utf8PathBuf` is to `String`. So being the analog of `&String` is a little surprising.)